### ISSUE TITLE
Question for Heating (/WW) Own Program Switchtimes, EMS-ESP32 function and what is already implemented?

### ISSUE THREAD

**Issue by giovanne123** [emsesp/EMS-ESP32#810]
Hi, I was looking if it is already possible to already a) view and b) modify the daily switchtimes for heating (circle 1/2) by EMS-ESP32?

I found in EMS-ESP32 already:
![image](https://user-images.githubusercontent.com/16689453/207825582-cd7b0f7a-cfb5-4480-b776-851c4f9c9048.png)
![image](https://user-images.githubusercontent.com/16689453/207825835-4d5c8977-0f51-439d-b43d-dd0785696fd5.png)
![image](https://user-images.githubusercontent.com/16689453/207820827-afffced1-778b-4a0f-97cf-01e42c2e9210.png)
![image](https://user-images.githubusercontent.com/16689453/207820950-43275498-9bb4-49e0-8ec2-d0ac801b38fd.png)
![image](https://user-images.githubusercontent.com/16689453/207821043-9abf0f37-1502-4ad8-9efb-ececd9b9b33c.png)
![image](https://user-images.githubusercontent.com/16689453/207821138-3178b812-92b1-43fe-9234-d8bb09f5a7e9.png)

But it is only ONE, only for ON and only for Monday!?
What about with OFF and the other weekdays?

In Buderus Software in the past I could see in graphical or table view (info: does not correlate with above EMS-ESP32 values because older screengrab):
HC1:
![image](https://user-images.githubusercontent.com/16689453/207827024-e49856a2-053b-487a-869d-6a8f8ad98fe5.png)

HC2:
![image](https://user-images.githubusercontent.com/16689453/207827111-a9d68d0b-4996-44f6-bf38-17e1b208f6b0.png)

I'm not sure if it is related e.g. for HC1 to "Telgram 0x3F"? 
-> HC1: https://emswiki.thefischer.net/doku.php?id=wiki:ems:telegramme#hk1schaltzeiten
-> HC2: (Typ 0x49)
-> "Schaltzeiten allgemein": https://emswiki.thefischer.net/doku.php?id=wiki:ems:telegramme#schaltzeiten_allgemein

**
Can this be also realized by EMS-ESP32?
- first to have a view with current own program settings (e.g. only in table view)
- second to be able to edit the settings by EMS-ESP32

or can you describe what is already implemeted in EMS-ESP32 and how it is working?**

And, maybe (but not so important for me) also for WW switchtimes?
![image](https://user-images.githubusercontent.com/16689453/207827607-f59bf77c-bc6d-40dd-9b06-43cd3e846d93.png)

**Comment by proddy** [comment]
There's some info in https://github.com/emsesp/EMS-ESP32/issues/182#issuecomment-967149449

**Comment by MichaelDvP** [comment]
There were a lot of discussions about that in different issue# for RC30, RC20, RC300, Junkers... 
Now it's poorly implemented for RC20/30/35. Send `no` only to read an entry, send full data to change an entry.
- We can not publish all entries in the mqtt, this blows the buffer. 
- also showing all on the thermstat webpage is to large
- the settings are different for other thermostats, so we need extra settings pages for each thermostat model
- same for holiday modes/dates

**Comment by giovanne123** [comment]
@proddy 
Thanks, yes there are some info in the comment you linked to and your last comment (when closing) in the thread is, that it should be resoved in 3.4. So I think that is the state I have found in 3.5.0b10 and described above.

For me in a first step it would be enough to see the current on/off's of programm own1.
For a short look e.g. in HA to see what times currently the heating will go on/off.
(so that I do not have to scroll to the RC30 controller itself. That was to complicated for me in the past ;-) I used the Buderus Ecosoft to view and change the on/off's before - but replaced the Ecosoft by EMS-ESP32 :-))

No need for me to change from EMS-ESP32 so far, because I think I also haven't understud fully how the current implementation works... have to dig deeper into the linked thread...

@MichaelDvP , ok thanks for the infos.
So seems not so simple.
Maybe we can get the important part (or telegram data) somehow to HA and the encryption of the data needs to be done on that end?

But maybe to complicated for the effort and because the on/off's are not that often changed...

... then I will write the on/off times manually to a HA table ;-) - but therefore (without Ecosoft) I have to scroll to RC30 controller :-(
